lib/citriage.rb

```diff
@@ -74,39 +79,70 @@ def list_jobs json
       mod_status = true
       failed_job = String.new
       begin
-        json['jobs'].each do |job|
-          if job['color'] == 'red'
-            mod_status = false
-            failed_job = job['url']
-            break
-          else
-            mod_status = true
+        json.each do |name, branch|
```

Comment:
Ah yeah, I'm totally on board with that. Having it mirror Jenkins is the idea anyway. This is a pattern that has always given me heartburn, so I look forward to seeing your solution!